src/components/textbox/textbox.pw.tsx

```diff
@@ -0,0 +1,1244 @@
+    );
+
+    const help = getDataComponentByValue(page, "help");
+
+    await expect(help).toHaveAttribute("aria-label", CHARACTERS.STANDARD);
+  });
+
+  (["left", "right"] as TextboxProps["align"][]).forEach((align) => {
+    test(`should render with align prop set to ${align}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TextboxComponent align={align} />);
+
+      await expect(textboxInput(page)).toHaveCSS("text-align", align as string);
+    });
+  });
+
+  test("should render with inputRef prop and focus on input via click on ref", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TextboxComponentInputRef />);
+
+    await getDataComponentByValue(page, "button").click();
+
+    await expect(textboxInput(page)).toBeFocused();
+  });
+
+  testData.forEach((value) => {
+    test(`should render and input ${value} as value`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TextboxComponent />);
+
+      await textboxInput(page).fill(value);
+
+      await expect(textboxInput(page)).toHaveValue(value);
+    });
+  });
+});
+
+test.describe("Event checks", () => {
+  test("should call onChange callback when a type event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    const inputValue = "1";
+    let callbackCount = 0;
+    await mount(
+      <TextboxComponent
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await textboxInput(page).type(inputValue);
+
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should call onBlur callback when a blur event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <TextboxComponent
+        onBlur={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await textboxInput(page).click();
+    await textboxInput(page).blur();
+
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should call onClick callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <TextboxComponent
+        onClick={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await textboxInput(page).click();
+
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should call onFocus callback when a focus event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <TextboxComponent
+        onFocus={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await textboxInput(page).focus();
+
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should call onMouseDown callback when a mousedown event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <TextboxComponent
+        onMouseDown={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await textboxInput(page).click({ button: "left" });
```

Comment:
**comment**: I'm not sure this is doing what we want - from the Playwright docs it seems that the `button` option defaults to `left` anyway (as you would expect), so this is the same as just calling `click()`.

To simulate a mousedown event it looks like you can use `page.mouse.down()` - see https://playwright.dev/java/docs/api/class-mouse#mouse-down . Although note that you probably first need to use the `move` method to get the mouse over the textbox - this can likely be done by using `getBoundingClientRect` but I'm beginning to think this is more complexity than it's worth and maybe the whole test can just be removed...

Reply:
LGTM!